crates/hir/src/lib.rs

```diff
@@ -107,13 +107,16 @@ pub use {
     hir_def::{
         adt::StructKind,
         attr::{Attr, Attrs, AttrsWithOwner, Documentation},
+        body::{Body, BodySourceMap},
         builtin_attr::AttributeTemplate,
+        expr::Expr,
         find_path::PrefixKind,
         import_map,
         nameres::ModuleSource,
         path::{ModPath, PathKind},
         type_ref::{Mutability, TypeRef},
         visibility::Visibility,
+        DefWithBodyId,
     },
```

Comment:
This list is preceded with "please don't add items here", which makes me unsure if it's okay.

Reply:
Yeah, the intention is for HIR to provide a facade to the internal APIs, so these four types are ones we wouldn't want to expose here. It'd be better to provide `is_inside_unsafe` on `Semantics`.

Reply:
Okay, done 👍🏻